imap_processing/tests/ena_maps/test_corrections.py

```diff
@@ -0,0 +1,231 @@
+"""Test coverage for ena_maps.corrections module."""
+
+from unittest import mock
+
+import numpy as np
+import pytest
+
+from imap_processing.ena_maps.utils.corrections import PowerLawFluxCorrector
+
+
+@pytest.fixture
+def hi_coeffs_file(ena_maps_test_data_path):
+    """Define the location of the hi coefficients file."""
+    return (
+        ena_maps_test_data_path
+        / "imap_hi_90sensor-esa-eta-fit-factors_20240101_v001.csv"
+    )
+
+
+@pytest.fixture
+def lo_coeffs_file(ena_maps_test_data_path):
+    """Define the location of the hi coefficients file."""
+    return ena_maps_test_data_path / "imap_lo_esa-eta-fit-factors_20240101_v001.csv"
+
+
+class TestPowerLawFluxCorrector:
+    """Test suite for ena_maps.corrections.PowerLawFluxCorrector."""
+
+    def test_load_coefficients_invalid_file(self, tmp_path):
+        """Test that loading a missing CSV file raises FileNotFoundError."""
+
+        with pytest.raises(FileNotFoundError):
+            PowerLawFluxCorrector(tmp_path / "missing.csv")
+
+    def test_eta_esa_non_negative(self, lo_coeffs_file):
+        """Test eta_esa will not return negative values."""
+
+        corr = PowerLawFluxCorrector(lo_coeffs_file)
+        k = np.array([1, 2])
+        # Experimentally found that gamma=10 produces negative eta
+        gamma = np.array([1.5, 10])
+        eta = corr.eta_esa(k, gamma)
+        assert eta[1] == 1
+
+    def test_estimate_power_law_with_uncertainties(self):
+        """Test slope estimation with flux uncertainties."""
+
+        fluxes = np.array([10, 20, 40, 80, 160, 320, 640])
+        energies = np.arange(8) + 1
+        uncertainties = np.sqrt(fluxes)
+        corr = PowerLawFluxCorrector.__new__(PowerLawFluxCorrector)
+        gamma, delta_gamma = corr.estimate_power_law_slope(
```

Comment:
This seems like an odd pattern, why not initialize it right away `PowerLawFluxCorrect().estimate_power_law_slope(...)`, or make this a static method if self isn't needed/used.